crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1219,6 +1219,158 @@ impl FromReflect for Cow<'static, str> {
     }
 }
 
+impl<T: FromReflect + Clone> List for Cow<'static, [T]> {
+    fn get(&self, index: usize) -> Option<&dyn Reflect> {
+        self.as_ref().get(index).map(|x| x as &dyn Reflect)
+    }
+
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect> {
+        self.to_mut().get_mut(index).map(|x| x as &mut dyn Reflect)
+    }
+
+    fn len(&self) -> usize {
+        self.as_ref().len()
+    }
+
+    fn iter(&self) -> crate::ListIter {
+        crate::ListIter::new(self)
+    }
+
+    fn drain(self: Box<Self>) -> Vec<Box<dyn Reflect>> {
+        #[allow(clippy::unnecessary_to_owned)]
```

Comment:
Might be worth noting why this is needed (clippy false positive).

Reply:
Good idea! I added a comment explaining how the `to_owned()` call can avoid cloning when you have a `Cow::Owned`.